### ISSUE TITLE
HEAD response must not include body

### ISSUE THREAD

**Issue by woonsan** [mar10/wsgidav#123]
Hi,

When I make a request with HEAD method like the following example, it includes body entity. It doesn't conform with RFC 2616 which says in Section 9.4: 

> The HEAD method is identical to GET except that the server MUST NOT
   return a message-body in the response

Could you please fix the issue by not including body entity in HEAD responses? Thanks in advance!

Here's an example output:

```
$ curl -i -XHEAD http://localhost:8888/folder1/
HTTP/1.1 200 OK
Content-Type: text/html
Date: Tue, 14 Aug 2018 15:44:32 GMT
Content-Length: 357
Server: WsgiDAV/2.4.1 Cheroot/6.4.0 Python/2.7.10

<!DOCTYPE HTML PUBLIC '-//W3C//DTD HTML 4.01//EN' 'http://www.w3.org/TR/html4/strict.dtd'>
<html><head>
  <meta http-equiv='Content-Type' content='text/html; charset=UTF-8'>
  <title>200 OK</title>
</head><body>
  <h1>200 OK</h1>
  <p>200 OK</p>
<hr/>
<a href='https://github.com/mar10/wsgidav/'>WsgiDAV/2.4.1</a> - 2018-08-14 11:44:32.666774
```

**Comment by mar10** [comment]
Thanks for reporting, I'll fixed this for 3.x.
How did this error affect you, i.e. how did you find it?

**Comment by woonsan** [comment]
Thanks a lot, @mar10 ! I will try with the newer version once ready.
While contributing to https://issues.apache.org/jira/browse/VFS-360, I used this software as backend in testing. If Keep-Alive is option is not used from client, there's no problem. If Keep-Alive is used, meaning old used socket connection being reused, the consecutive request fails because the response was polluted by the previous head response. In VFS2 with my new HttpClient v4 code, Keep-Alive is now the default option, so I had to hit this issue.
Anyway, this is a great project! Cheers!
